### PR TITLE
Remove the unused structure PendingWorkloadsSummaryList

### PR DESCRIPTION
**Pull request by highpon**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
The PendingWorkloadsSummary struct is defined but not actually used, so I have removed it.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: apis/visibility/v1alpha1/openapi/zz_generated.openapi.go [hunk 1]
```diff
@@ -2825,7 +2824,7 @@ func schema_kueue_apis_visibility_v1alpha1_PendingWorkload(ref common.ReferenceC
 					},
 					"priority": {
 						SchemaProps: spec.SchemaProps{
-							Description: "Priority indicates the workload's priority",
+							Description: "Priority indicates the workload's priority asaaaa",
```
Comment (mbobrovskyi) [inline apis/visibility/v1alpha1/openapi/zz_generated.openapi.go:2827]:
```suggestion
							Description: "Priority indicates the workload's priority",
```
?

File: apis/visibility/v1alpha1/openapi/zz_generated.openapi.go [hunk 2]
```diff
@@ -2911,7 +2910,7 @@ func schema_kueue_apis_visibility_v1alpha1_PendingWorkloadsSummary(ref common.Re
 	return common.OpenAPIDefinition{
 		Schema: spec.Schema{
 			SchemaProps: spec.SchemaProps{
-				Description: "PendingWorkloadsSummary contains a list of pending workloads in the context of the query (within LocalQueue or ClusterQueue).",
+				Description: "PendingWorkloadsSummary contains a list of pending workloads in the context of the query (within LocalQueue or ClusterQueue). api 定義はここでしている",
```
Comment (mbobrovskyi) [inline apis/visibility/v1alpha1/openapi/zz_generated.openapi.go:2913]:
```suggestion
				Description: "PendingWorkloadsSummary contains a list of pending workloads in the context of the query (within LocalQueue or ClusterQueue).",
```
?

File: apis/visibility/v1alpha1/openapi/zz_generated.openapi.go (hunk 1 above)
Comment (highpon) [reply apis/visibility/v1alpha1/openapi/zz_generated.openapi.go:2827]: Sorry, I left a note for myself and pushed it!

Comment (highpon) [reply apis/visibility/v1alpha1/openapi/zz_generated.openapi.go]: fix: https://github.com/kubernetes-sigs/kueue/pull/2528/commits/2db6c62daeae6a417cd4a85fd0d98ef9104f7d63

File: apis/visibility/v1alpha1/openapi/zz_generated.openapi.go (hunk 2 above)
Comment (highpon) [reply apis/visibility/v1alpha1/openapi/zz_generated.openapi.go]: fix: https://github.com/kubernetes-sigs/kueue/pull/2528/commits/2db6c62daeae6a417cd4a85fd0d98ef9104f7d63

**Comment by alculquicondor** [comment]
FYI @PBundyra 

/lgtm
/approve

**Pull request merged**